media.go

```diff
@@ -501,6 +554,10 @@ func (ms *mediaSource[T, U]) Stream(ctx context.Context, errHandlers ...ErrorHan
 	prodCon.stateMu.Lock()
 	defer prodCon.stateMu.Unlock()
 
+	if currentSpan := trace.FromContext(ctx); currentSpan != nil {
+		prodCon.cancelCtx = trace.NewContext(prodCon.cancelCtx, currentSpan)
+	}
+
```

Comment:
Why is this needed? Why not just `StartSpan` at the top of this function?

I'm also not seeing any info in our trace logs from this. If we get this working we'd also want to add a span for its close function.

Reply:
This is a similar story to the last one, we needed the span from `ctx` to be put into `prodCon.cancelCtx` so it could be used in functions which don't accept a `context.Context` but do use `prodCon.cancelCtx`. This construction wasn't meant to start a span, just pass the current one through, which is why it doesn't show up in the trace logs. I put it so low in the function because it needed to be after `prodCon` was given its `cancelCtx` and I assumed the `prodCon.stateMu` concerns the `cancelCtx` as well

Reply:
Oh, I see. Thanks for the explanation. I was going to suggest adding a span to time this method, but there are three different contexts in this function, so figuring out which to use and how may be a bit more hassle than it's worth especially considering this may be refactored soon. I like what you've done here. This seems like the simplest solution for now. LGTM!!